### PR TITLE
CRM-19353 - 'Dedupe by email' not working 2

### PR DESCRIPTION
**Pull request by hakuren**
[CRM-19353: 'Dedupe by email' not working](https://issues.civicrm.org/jira/browse/CRM-19353)

### CODE REVIEW

**Comment by colemanw** [comment]
@civicrm-builder add to whitelist

**Review by colemanw — CHANGES_REQUESTED** [review]
This needs a couple of [style fixes](https://test.civicrm.org/job/CiviCRM-Core-PR/11828/checkstyleResult/new).

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by seamuslee001** [comment]
Jenkins re test this please

File: CRM/Contact/BAO/Query.php [hunk 1]
```diff
@@ -4335,7 +4336,10 @@ public static function apiQuery(
     $sql = "$select $from $where $having";
 
     // add group by
-    if ($query->_useGroupBy) {
+    if(!empty($groupBy)) {
```
Comment (seamuslee001) [inline CRM/Contact/BAO/Query.php]: @hakuren you need to do if ( not if(

**Comment by eileenmcnaughton** [comment]
We discussed this at the sprint & we had some concerns

1) this was adding a new parameter to the api & to the api architecture ('group_by') and we weren't comfortable with that being added in an undiscussed, untested way
2) We weren't convinced the UI implied that test recipients would be deduped

I'm going to close this PR because it is against an out of date RC.

I'd encourage you to open a new ticket for the outstanding issue around test recipients being deduped, since I think #9214 resolved CRM-19353 - (unless you no longer care)

Are you on the dev email list? It would be a good place to discuss adding group_by to the api?

**Pull request closed without merging**